icu/source/icudefs.cmake

```diff
@@ -120,7 +120,7 @@ if(WIN32)
 endif()
 
 # overridden by icucross.cmake
-if(MSVC)
+if(CMAKE_GENERATOR MATCHES "Visual Studio.*")
```

Comment:
Thank you for contributing!

Could you please check the building with CMake 3.3, Ninja and MSVC?

Nowhere can I find information about fixing this problem in the CMake 3.4.

I use Ninja 1.8.2 and MSVC 19.11.25547.0 from Visual Studio 15 2017.

With CMake 3.4 all is good. With CMake 3.3.0 or 3.3.2 I get the error:
```
ninja: error: build.ninja:2490: bad $-escape (literal $ must be written as $$)
```
And in the build.ninja:2490
```
E:\Dev\Tools\cmake\cmake-3.3.2-win32-x86\bin\cmake.exe -E copy_if_different E:/Dev/Temp/ICU_build/bin/stubdata/icudt$<$<CONFIG:Debug>:d>.dll E:/Dev/Temp/ICU_build/bin/"
```
In the CMake files the problematic line is [this](https://github.com/LibCMaker/ICU_CMake_Files/blob/master/icu/source/stubdata/CMakeLists.txt#L46) with "$<TARGET_FILE:${lib_NAME}>".

Reply:
> With CMake 3.4 all is good

... and with your changes.

Reply:
Hmmm not sure about it. While the `$<$<CONFIG:Debug>:d>` is more modern there could just be

```
if (CMAKE_BUILD_TYPE MATCHES DEBUG)
```

Is CMake 3.3 a hard requirement?

Reply:
If CMake 3.3, Ninja and MSVC are not working combination then I will add a restriction to CMake 3.4 for this combination. But first I wanted to get confirmation for this need.

> While the $<$<CONFIG:Debug>:d> is more modern there could just be

Please see [this explanation](https://stackoverflow.com/a/24470998).

> Is CMake 3.3 a hard requirement?

COMPILE_LANGUAGE generator expression was introduced in CMake 3.3 which is used in ICU_CMake_Files.

Reply:
I found an explanation in [CMake 3.4 Release Notes / Properties](https://cmake.org/cmake/help/v3.12/release/3.4.html#properties):
"The OUTPUT_NAME target property and its variants learned to support generator expressions."
It does not depend on the Ninja and MSVC.